dialect.go

```diff
@@ -24,26 +25,26 @@ type Dialect interface {
 	DataTypeOf(field *StructField) string
 
 	// HasIndex check has index or not
-	HasIndex(tableName string, indexName string) bool
+	HasIndex(ctx context.Context, tableName string, indexName string) bool
```

Comment:
All of these are public and yet you changed their signature. This is then fit for gorm v2 because of breaking changes. Your PR description says otherwise though.